src/games/relay/game.ts

```diff
@@ -23,37 +26,73 @@ export interface MyGameState {
   url: string;
 }
 
-const lengthOfCompetition = 60 * 60; // seconds
+function parseGameState(json: string): MyGameState {
+  const parsed = JSON.parse(json);
+
+  if (typeof parsed !== 'object' || parsed === null) {
+    throw new Error('relay: game_state_from_json: Invalid JSON: not an object');
+  }
+  return parsed as MyGameState;
+}
+
+function loadGameState(events: EventsAPI): MyGameState | null {
+  const phase = localStorage.getItem("RelayGamePhase");
+  const gameStateJSON = localStorage.getItem("RelayGameState");
+  if (phase === null || gameStateJSON === null) {
+    events.endTurn();
+    return null;
+  }
+
+  let state;
+  try { // if the json is bad, continue as if we didnt even have it
+    state = parseGameState(gameStateJSON);
+  } catch {
+    events.endTurn();
+    console.error("could not load game phase from json, invalid json");
+    return null;
+  }
 
-const GUESSER_PLAYER = '0';
-const JUDGE_PLAYER = '1';
+  events.setPhase(phase);
+  state.milisecondsRemaining = Date.parse(state.end) - Date.now();
+  return state;
+}
+const lengthOfCompetition = 60 * 60; // seconds
```

Comment:
The strategy has the same function as I understand, so I would merge them in a `utils.ts` file, and give a parameter as the game name. (It can be called as `GameRelay.name`) And probably you can store it only in one localstorage item and just get them from there.

Reply:
I will try to do that.